server/src/main/java/org/opensearch/cluster/routing/RoutingNodes.java

```diff
@@ -191,30 +192,39 @@ private void updateRecoveryCounts(final ShardRouting routing, final boolean incr
             return;
         }
 
-        Recoveries.getOrAdd(recoveriesPerNode, routing.currentNodeId()).addIncoming(howMany);
+        Recoveries.getOrAdd(getReplicaRecoveries(routing), routing.currentNodeId()).addIncoming(howMany);
 
         if (routing.recoverySource().getType() == RecoverySource.Type.PEER) {
             // add/remove corresponding outgoing recovery on node with primary shard
             if (primary == null) {
                 throw new IllegalStateException("shard is peer recovering but primary is unassigned");
             }
-            Recoveries.getOrAdd(recoveriesPerNode, primary.currentNodeId()).addOutgoing(howMany);
+
+            Recoveries.getOrAdd(getReplicaRecoveries(routing), primary.currentNodeId()).addOutgoing(howMany);
```

Comment:
At this point it can still be non-replica recoveries?. Is `getReplicaRecoveries ` most apt?

Reply:
Yes, you are right. I will rename.